native_client/deepspeech.h

```diff
@@ -128,8 +98,77 @@ namespace DeepSpeech
       char* stt(const short* aBuffer,
                 unsigned int aBufferSize,
                 int aSampleRate);
+
+      /**
+       * @brief Setup a context used for performing streaming inference.
+       *        the context pointer returned by this function can then be passed
+       *        to {@link feedAudioContent()} and {@link finishStream()}.
+       *
+       * @param aPreAllocFrames Number of timestep frames to reserve. One timestep
+       *                        is equivalent to two window lenghts (50ms), so
+       *                        by default we reserve enough frames for 3 seconds
+       *                        of audio.
+       * @param aSampleRate The sample-rate of the audio signal.
+       *
+       * @return A context pointer that represents the streaming state.
+       */
+      StreamingState* setupStream(unsigned int aPreAllocFrames = 150,
+                                  unsigned int aSampleRate = 16000);
```

Comment:
The more I think about that, the more I'm wondering if we should not have some graph metadata somehow to rely on this.

Reply:
aPreAllocFrames is just an optimization knob, so not tied to the graph. The sample rate is kind of tied, in that it should be larger than or equal to the sample rate of the training data. (This requirement might go away depending on the result of our band-pass experiments).